crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -655,7 +656,7 @@ impl AuthorityPerEpochStore {
         db_options: Option<Options>,
         metrics: Arc<EpochMetrics>,
         epoch_start_configuration: EpochStartConfiguration,
-        store: Arc<AuthorityStore>,
+        cache_reader: Arc<ExecutionCache>,
```

Comment:
Should this be ExecutionCacheRead if it's a cache_reader?

Reply:
It also needs to be interpreted as a `&dyn ObjectStore`, so it turned out to be too difficult to use a trait here instead of a concrete type.